### PR TITLE
feat(ui)!: add email verification and allow to unlink social providers from profile screen

### LINKED ISSUE #7609: SignScreen flutterfire ui : Register
After registering through email and password, the normal process flow should be to sign out the user and send a verification email then send the user  to the sign in screen. This will be more helpful.

### PR DESCRIPTION
**Pull request by lesnitsky**
## Description

This PR implements email verifiaction and provider unlinking

## Related Issues

Closes #7609

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

> `actionCodeSettings` parameter was dropped from the `EmailProviderConfiguration` since it is no longer used

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/auth/widgets/error_text.dart [hunk 1]
```diff
@@ -19,6 +19,9 @@ String? localizedErrorText(
       return labels.accessDisabledErrorText;
     case 'wrong-password':
       return labels.wrongOrNoPasswordErrorText;
+    case 'credential-already-in-use':
+      // TODO(lesnitsky): add translation
+      return 'This credential is already associated with a different user account.';
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/auth/widgets/error_text.dart:24]: This and other hardcoded strings are going to be addressed in a separate localisation PR

**Review by Salakar — APPROVED** [review]

File: packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        const Align(
+          child: Title(text: 'Verify your email'),
+        ),
+        const SizedBox(height: 32),
+        const Text(
+          'A verification email has been sent to your email address. '
```
Comment (Salakar) [inline packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart:125]: Guessing this and other texts should be localized in that future PR right?

Comment (lesnitsky) [reply packages/flutterfire_ui/lib/src/auth/screens/email_verification_screen.dart:125]: https://github.com/firebase/flutterfire/pull/8358#discussion_r837589464

File: packages/flutterfire_ui/lib/src/auth/email_verification.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  Future<void> sendVerificationEmail(
+    TargetPlatform platform,
+    ActionCodeSettings? actionCodeSettings,
+  ) async {
+    value = EmailVerificationState.sending;
+    try {
+      await user.sendEmailVerification(actionCodeSettings);
+    } on Exception catch (e) {
+      error = e;
```
Comment (russellwheatley) [inline packages/flutterfire_ui/lib/src/auth/email_verification.dart:52]: Is there a worry this error could be smothered if there is an error in the next try/catch?

**Review by russellwheatley — CHANGES_REQUESTED** [review]
couple of questions 🤓

File: packages/flutterfire_ui/lib/src/auth/email_verification.dart (hunk 3 above)
Comment (lesnitsky) [reply packages/flutterfire_ui/lib/src/auth/email_verification.dart:52]: oh, actually if there is an exception in this phase there is no point in proceeding

File: packages/flutterfire_ui/lib/src/auth/email_verification.dart [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      value = EmailVerificationState.failed;
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/auth/email_verification.dart:53]:
```suggestion
      value = EmailVerificationState.failed;
      return nulll;
```

File: packages/flutterfire_ui/example/lib/main.dart [hunk 5]
```diff
@@ -64,8 +84,12 @@ class FirebaseAuthUIExample extends StatelessWidget {
         inputDecorationTheme: const InputDecorationTheme(
           border: OutlineInputBorder(),
         ),
+        elevatedButtonTheme: ElevatedButtonThemeData(style: buttonStyle),
+        textButtonTheme: TextButtonThemeData(style: buttonStyle),
+        outlinedButtonTheme: OutlinedButtonThemeData(style: buttonStyle),
       ),
-      initialRoute: auth.currentUser == null ? '/' : '/profile',
+      // theme: const CupertinoThemeData(brightness: Brightness.light),
```
[the remark this answers is not in the document]
Comment (lesnitsky) [reply packages/flutterfire_ui/example/lib/main.dart:91]: yeah, I was just checking if everything works fine with CupertinoApp

Comment (lesnitsky) [inline packages/flutterfire_ui/example/lib/main.dart:91]:
```suggestion
```

**Review by russellwheatley — APPROVED** [review]
LGTM

**Pull request merged**